xonsh/aliases.py

```diff
@@ -73,6 +74,20 @@ def source_bash(args, stdin=None):
     return
 
 
+def xexec(args, stdin=None):
+    """
+    Replaces current process with command specified and passes in the
+    current xonsh environment.
+    """
+    env = builtins.__xonsh_env__
+    denv = env.detype()
+    if (len(args) > 0):
+        try:
+            os.execvpe(args[0], args, denv)
+        except FileNotFoundError as e:
+            return "xonsh: " + e.args[1] + ": " + args[0] + "\n"
```

Comment:
Any preference on what that should be? Otherwise I'll just make it `xonsh: exec: no args specified"